### ISSUE TITLE
Suboptimal error using mutate

### ISSUE THREAD

**Issue by jeffreypullin** [tidyverse/dplyr#4780]
Hi, 

I was trying out some of the `rowwise` stuff in the the dev version when I hit the following error message, parts of which I thought could be a bit clearer. 

``` r
library(dplyr, warn.conflicts = FALSE)

parameters <- tibble::tribble(
  ~mu, ~sd,
    1,   2,
   10,  20,
)

parameters %>%
  rowwise() %>%
  mutate(data = rnorm(10, mu, sd))
#> Error: `mutate()` argument `data` must be recyclable.
#> ℹ `data` is rnorm(10, mu, sd)
#> x `x` can't be recycled to size 1.
#> x It must be size 1 or 1, not 10.
```

<sup>Created on 2020-01-19 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

When I was reading the error message: 

- I was very confused about what `x` is as there is no `x` anywhere in the call.
- I found the `1 or 1` clause quite odd and had to reread it a couple of times to firstly make sure that I had not misread and then work out why it was like that. 

Just thought these might be useful observations - feel free to close if not. 

All the best!

**Comment by romainfrancois** [comment]
In this PR https://github.com/tidyverse/dplyr/pull/4746 things might be a little better, however we've only just rehabilitated `rowwise` from questioning state, so indeed we need to take some more care about error messages in the `rowwise` case.

**Comment by hadley** [comment]
@romainfrancois I think we're also missing a general fix for `size 1 or 1`, and in `rowwise()` maybe we should specifically hint to use `list()`? Where is the `x` in the error coming from?

**Comment by romainfrancois** [comment]
It currently looks like this in #4746 

``` r
library(dplyr, warn.conflicts = FALSE)

parameters <- tibble::tribble(
  ~mu, ~sd,
    1,   2,
   10,  20,
)

parameters %>%
  rowwise() %>%
  mutate(data = rnorm(10, mu, sd))
#> Error: `mutate()` argument `data` must be recyclable.
#> ℹ `data` is `rnorm(10, mu, sd)`.
#> x `data` can't be recycled to size 1.
#> ℹ `data` must be size 1, not 10.
```

<sup>Created on 2020-01-20 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9000)</sup>

I'll add rowwise specific things in a follow up PR.

**Comment by romainfrancois** [comment]
closed by #4783
